### ISSUE TITLE
snyk auth doesn't authenticate and errors out

### ISSUE THREAD

**Issue by cfanoulis** [snyk/snyk#220]
- `node -v`: Latest
- `npm -v`: Latest
- `snyk -v`: Latest
- Command run: snyk auth

### Expected behaviour
The CI can be authorized... somehow

### Actual behaviour
I have tried everything but the CI machine is not authenticating

### Steps to reproduce
I have tried to authenticate with every way possible in the CircleCI (set environment for command/set global in container/set project global, see https://github.com/cfanoulis/hulabot/pull/12).

---
Error trace:
```
#!/bin/bash -eo pipefail
snyk auth
An unknown error occurred. Please include the trace below when reporting to Snyk:

Error: noAuthInCI
    at githubAuth (/usr/local/lib/node_modules/snyk/cli/commands/auth.js:45:17)
    at auth (/usr/local/lib/node_modules/snyk/cli/commands/auth.js:155:15)
    at /usr/local/lib/node_modules/snyk/lib/hotload.js:16:21
    at Object.<anonymous> (/usr/local/lib/node_modules/snyk/cli/index.js:26:23)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)

Exited with code 1
```

**Comment by adrukh** [comment]
Thanks for reporting this @cfanoulis !

You can authenticate the CLI in one of two ways:
* set the `SNYK_TOKEN` env var to the value of your api token
* `snyk auth <your api token>` - this validates the token and places it in the local config

If you choose the env var, you do *not* need to run `snyk auth <api token>`, you can run `snyk test` or any other command.

From the output above it seems like you are running `snyk auth` without an api token, which triggers a manual authentication flow (opens a browser window for you to authenticate), and is not supported in CI envs for lack of user interaction.

Please clarify which way are you trying to authenticate?

**Comment by cfanoulis** [comment]
Oh haha! I am trying to authenticate even when I set the env variables
Shoot! Let me try again. I will report in ~1 minute

**Comment by cfanoulis** [comment]
Mmhm. Now it asks for an authentication.

[CI link](https://circleci.com/gh/cfanoulis/hulabot/135?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

Error trace:
```
#!/bin/bash -eo pipefail
npm run snyk

> hulabot@1.0.0 snyk /home/circleci/project
> snyk test && snyk monitor

snyk test requires an authenticated account. Please run `snyk auth` and try again.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! hulabot@1.0.0 snyk: `snyk test && snyk monitor`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the hulabot@1.0.0 snyk script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/circleci/.npm/_logs/2018-09-20T17_52_36_350Z-debug.log
Exited with code 1
```

**Comment by adrukh** [comment]
Seems like the env var is not propagating properly into the environment?
Try adding something like `echo ${SNYK_TOKEN:0:4}` to have part of your api token printed. If you get an empty string, the env var isn't available...

**Comment by cfanoulis** [comment]
:joy: I cannot believe this. The env was `$SNYK_KEY`, which is the name of the project env key. If I was to rename that variable, would it work?

**Comment by adrukh** [comment]
I'm willing to bet it will!

**Comment by cfanoulis** [comment]
CI has started...

**Comment by cfanoulis** [comment]
YES! IT FINALLY WORKED! Thank you very much adrukh, if you wouldn't mind I am closing this

**Comment by adrukh** [comment]
Glad to have helped :)

**Comment by sandeepkkv** [comment]
Hi, I'm trying to run snyk auth from my CLI, but it gives me the error "Unable to get local issuer certificate". I have attached the snapshot of the same. 
Thanks in Advance.
![WhatsApp Image 2019-05-05 at 14 08 20](https://user-images.githubusercontent.com/32559505/57200500-b151db80-6f41-11e9-93bd-a371d45cff10.jpeg)

**Comment by tuf22191** [comment]
@adrukh 's comment is good. Just for clarification: 
1. Go here: https://app.snyk.io/account
2. click to see your token value under the API token section
3. copy this token and then set the SNYK_TOKEN="" environment variable depending on if you are on Windows or Linux

**Comment by alohe** [comment]
If anyone is having trouble authenticating from a server 
`snyk auth <your api token>`
This worked for me.

**Comment by lucatsf** [comment]
1. Go here: https://app.snyk.io/account
2. snyk auth <your api token>

following these steps that were shared by [tuf22191](https://github.com/tuf22191) and  [Kyslik](https://github.com/Kyslik) worked for me
